### ISSUE TITLE
New plugin called "JetMembership"

### ISSUE THREAD

**Issue by Lonsdale201** [Crocoblock/suggestions#968]
**Introductory.** 

WIth this new plugin, you can open unlimited possibles to the Jetengine

With the Jetengine you can create front end sub, but would be more effective if you can controlling with JetMembership.

**Short description.**

Many type of Membership to earn money with submission, calm listing, and possible to do more automatized system, with periodic billing options, and set advanced rules.

**Core features:**

- Membership types

- Membership rules

- Billing Cycles

- Payment options

- Membership card widget for the elementor

- Full compatibility with the Jetengine profile builder

**When you create the membertype, you can add:**

- [ ] Member type name or package name.

- [ ] Advanced rules to set, how many post can submit

- [ ] How many pictures, or document or video can attach. ( number and size limit too)

- [ ] What post types including for this Membertype.

- [ ] Featured post options with controlling this is, below the Listing widget.

**Additional Membertype settings:**

Set this Membertype free or paid, limited period.
Set the billing Cycles
If the billing cycles selected Period times, what happening if the next time not paid.
The user delete temporarily, their post automatice going to draft, or full delete.
 
**Billing Cycles**

Pay one time.
Pay with period times.
Pay in specific date.

Payment method(s):
Paypal, Stripe, and Woocommerce compatibility.

**Comment by shenom-x** [comment]
Yes please!

We´re looking for that kind solution right now. In our case it must be a mix of membership and booking. Our brand partners should book simple advertising spaces on our page - that they can administrated by their selfs.

Book, pause, cancel and upload their ad images and text claims as well.

**Comment by gergelias5** [comment]
Absolutely. This would be the ideal plugin to complement JetEngine. Very much needed.

**Comment by Samurhan** [comment]
Pleeeeeeeease

**Comment by eybel** [comment]
Yes! On the same boat I came here to open a new feature request but I saw this one.

JetMembership for User Profile and compatible.

My idea: a Membership post type can be created with JetEngine and its values too!

But then "User Profile" should check for this membership post type and its optionis before updating, submitting, logging in, etc...

In my case I have multiple STAFF members that will have this USER PROFILE page and they should be able to do things or not based on Memberships.

**Comment by DmitryBychenko** [comment]
You already have show / hide functionality for logged-in & logged out users.
Please, add these features:

1. Section / Column / Widget visibility control for specific **Membership levels** (NOT **Roles**, i.e. Admin, Editor, Customer, but Silver, Gold, Platinum, VIP **Levels**) in Elementor. 
There are 2 awesome plugins - [Groups](https://wordpress.org/plugins/groups/) by itthinx, and [Members](https://wordpress.org/plugins/members/) by MemberPress.
These plugins do great job, maybe just JetEngine integration with these plugin is enough?

2. Integration with WooCommerce to accept payments. 
Maybe other users want to use built-in gateways, like PayPal, Stripe...etc.
So make them disable-able.
Turn off WooCommerce compatibility if you dont need it, OR turn off built-in gateways.

3. Assign specific **Membership level** (NOT Roles, i.e. Admin, Editor, Customer) after purchase specific WooCommerce Product / pay through built-in gateways.

4. Support for Variable Products. I.e. if Product is variable, each variation should open its own Membership Level.

5. Shortcodes and / or Gutenberg blocks to show / hide part of content on the page in native Gutenberg editor. 
**Including** _Show content to all levels (& logged-in users), except specific level_.

6. Show / Hide menu items depends on user role: 
https://www.mangoblogger.com/wp-content/uploads/2017/12/nav_menu1.png 
or 
https://www.mangoblogger.com/wp-content/uploads/2017/12/If_Menu1.png
or something similar.
Maybe integration with JetBlocks plugin.

7. Compatibility with WooCommerce Subscriptions or other subscription plugins to accept recurring payments.
Or make JetMembership capable to accept recurring payments.

**Comment by lolirado** [comment]
I just create the membership site half crocoblock and half other membership plugin.
Imagine if we can now 100% Crocoblock

Let's do it !!!

**Comment by Willitbang** [comment]
Please make this plugin. This would take JetEngine to the next level!

**Comment by maurocarreraok** [comment]
Would be amazing.

**Comment by rajas91** [comment]
This would be really awesome thing to have

**Comment by sebak2003** [comment]
Yes please! With "Woocommerce Subscription" Compatibility. 

- Feature: Drip content (Access to post at x week or month after subscription date)

Please! To not use anymore Woocommerce Membership 👍

**Comment by bluezmedia** [comment]
Vote up too !

**Comment by rajas91** [comment]
+1 , 
This functionality is much needed

**Comment by dsmoina** [comment]
I absolutely love JetEngine and the Jet product line! This functionality would really take everything to the next level. :)

**Comment by petranca** [comment]
+1

**Comment by TomEbeyer** [comment]
+1

**Comment by berbaherz** [comment]
Please a chat function too, that would make it a game changer. So we were supposed to get it this year, when are we getting it?

**Comment by futureyoon** [comment]
+1

**Comment by marcel-za** [comment]
+1

**Comment by KHRifat** [comment]
Great, I am looking for this solution. I have to build 2 projects, I expect this would be released very soon.

**Comment by maurocarreraok** [comment]
+1

**Comment by julianohammerle** [comment]
+1

**Comment by SaltedCherry** [comment]
That will be fantastic!

**Comment by imCorfitz** [comment]
This is exactly what I am looking for! I am using another plugin at the moment, but would love to have as much as possible under the "JetUmbrella" so to speak. +1 from here...

**Comment by imCorfitz** [comment]
Let me just second this one again and let you know I am willing to join your team for FREE in the development of this as I need it NOW :D

**Comment by KHRifat** [comment]
+1

JetMembership would complete the Crocoblock's Stack... So fulfill you vault @Crocoblock

**Comment by ayhanmalkoc** [comment]
+1

**Comment by zanga7** [comment]
Im trying to achieve this with Jet Engine Profile Builder and JetWooBuilder but the two do not work together :-(
Perhaps that should be fixed first.

**Comment by DmitryBychenko** [comment]
Any news on what developers think about creating this plugin?

**Comment by Roberto77** [comment]
Please give us news. This plugin is very important!

**Comment by EvgenDoman** [comment]
+1

**Comment by ayhanmalkoc** [comment]
+1

**Comment by xC4M** [comment]
Yes, please!!!!!!! 🙌🏼🙌🏼🙌🏼

**Comment by puertarodrigo** [comment]
+1

**Comment by SchneiderSam** [comment]
+1 oh yes. Would be a game changer

**Comment by WaBaX** [comment]
+1, 

I need this function very urgently and would therefore also contribute to a limited extent to the development costs.

Are there other interested parties for this function,
who would contribute to the development costs to speed it up?

Should this function actually be implemented,
do I need a REST API / webhook option, to connect my accounting system, 
which can generate then the invoices automatically.
Access to all meta fields will be necessary for this.

I would be very grateful for a comment from CrocoBlock on the current status of this request.

**Comment by BaronVonDerpsalot** [comment]
+1

**Comment by GCG82** [comment]
+1

**Comment by ofmarconi** [comment]
+1 can't miss sending and receiving Webhook, to integrate with other platforms.

For example create users by receiving a Webhook (and also update existing user information)

**Comment by ofmarconi** [comment]
@Lonsdale201 
 > JetMembership

How did you solve this need?

**Comment by gonzalogcaminos2010** [comment]
Yes please!!! +100

**Comment by neylwalecki** [comment]
+1

**Comment by davidwatkins2** [comment]
Yes, please! I hope Crocoblock listens to their customer’s feedback because most companies don't. 

Also,

I believe having a shortcode that could hide specific content from users is essential too. 

For Example:

A user could read a blog, but is stopped with a “Read More” button that when a user clicks on the button directs them to a signup page of the relating subscription that would give them access to the rest of the content. 

Here is an example with Restrict Content Pro
https://help.ithemes.com/hc/en-us/articles/360049407014--restrict-

Here is text from their website explaining how it works

Restricting Portions of Content

“Instead of automatically restricting the entire contents of a post/page with the meta box option, you can also restrict only portions of the content with a shortcode. This method is more flexible and is best suited for users who want to block off sections of their content to subscribers, while leaving other blocks open to all users” Rsstrict Content Pro

Also other important features would be
1. Content Drip (content and page scheduling availability by date & time)
2. Member levels
3.  Allowing Members to upgrade (paying the difference) or downgrade ( receive partial/ prorated refund within a certain amount of days and if that amount of days passed already the member can still downgrade without the refund ) 
4. Ability to  completely  customize signup forms (Member Press & Restrict Content Pro) doesn't allow this. They only allow you to Add form fields.

**Comment by developer784** [comment]
+1 asap...

**Comment by wandersonce** [comment]
+1 waiting...

**Comment by Ellinasg** [comment]
+1

Where's this at currently?

**Comment by bbua97** [comment]
+1

**Comment by catchingclicks** [comment]
+1 asap!

**Comment by zup44** [comment]
+100000 asap!

**Comment by sakibstime** [comment]
+1

**Comment by magdakubis** [comment]
Yes please!!!

**Comment by TheBigBanik** [comment]
At least, can the Croco team reply if this is something they are thinking of or we will just wait like for ever? This plugin will make the Crocoblock the ultimate company in wordpress. +10000000000000000000 to build it.
